SYCL/ESIMD/api/functional/operators/operator_assignment_move_core.cpp

```diff
@@ -0,0 +1,61 @@
+//==------- operator_assignment_move_core.cpp  - DPC++ ESIMD on-device test
+//          ----------------------------------------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu, level_zero
+// XREQUIRES: gpu
+// TODO gpu and level_zero in REQUIRES due to only this platforms supported yet.
+// The current "REQUIRES" should be replaced with "gpu" only as mentioned in
+// "XREQUIRES".
+// RUN: %clangxx -fsycl %s -fsycl-device-code-split=per_kernel -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// XFAIL: *
+// TODO Remove XFAIL once the simd vector provides move assignment operator
+//
+// Test for simd move assignment operator.
+// The test creates source simd instance with reference data and invokes move
+// assignment operator from source to destination simd instance. It is expected
+// for destination simd instance to store a bitwise same data as the reference
+// one.
+
+#include "operator_assignment.hpp"
+
+using namespace sycl::ext::intel::experimental::esimd;
+using namespace esimd_test::api::functional;
+
+// Descriptor class for the case of calling move assignment operator.
+struct move_assignment {
```

Comment:
Nit: I would have move and copy ctors in the same source as they share a lot

Reply:
I've applied this comment in this commit: a80c2e39b68ed61f908a5454b77590dbc907f510